src/malli/swagger.cljc

```diff
@@ -67,3 +68,102 @@
                                  ::json-schema/definitions definitions
                                  ::json-schema/transform -transform})]
      (cond-> (-transform ?schema options) (seq @definitions) (assoc :definitions @definitions)))))
+
+(defn remove-empty-keys
+  [m]
+  (into (empty m) (filter (comp not nil? val) m)))
+
+(defmulti extract-parameter (fn [in _] in))
+
+(defmethod extract-parameter :body [_ schema]
+  (let [swagger-schema (transform schema {:in :body, :type :parameter})]
+    [{:in          "body"
+      :name        (:title swagger-schema "body")
+      :description (:description swagger-schema "")
+      :required    (not= :maybe (m/type schema))
+      :schema      swagger-schema}]))
+
+(defmethod extract-parameter :default [in schema]
+  (let [{:keys [properties required definitions]} (transform schema {:in in, :type :parameter})]
+    (println "\nextract-parameter definitions:"
+             (with-out-str (clojure.pprint/pprint definitions)))
+    (mapv
+      (fn [[k {:keys [type] :as schema}]]
+        (merge
+          {:in          (name in)
+           :name        k
+           :description (:description schema "")
+           :type        type
+           :required    (contains? (set required) k)}
+          schema))
+      properties)))
+
+(defmulti expand (fn [k _ _ _] k))
+
+#_(defn lift-response-definitions
+    [response]
+    (if-let [definitions (get-in response [:schema :definitions])]
+      (-> response
+          (update :schema dissoc :definitions)
+          (assoc :definitions definitions))))
+
+(defmethod expand ::responses [_ v acc _]
+  {:responses
+   (into
+     (or (:responses acc) {})
+     (for [[status response] v]
+       [status (-> response
+                   (update :schema transform {:type :schema})
+                   (update :description (fnil identity ""))
+                   remove-empty-keys)]))})
+
+(defmethod expand ::parameters [_ v acc _]
+  (let [old    (or (:parameters acc) [])
+        new    (mapcat (fn [[in spec]] (extract-parameter in spec)) v)
+        merged (->> (into old new)
+                    reverse
+                    (reduce
+                      (fn [[ps cache :as acc] p]
+                        (let [c (select-keys p [:in :name])]
+                          (if (cache c)
+                            acc
+                            [(conj ps p) (conj cache c)])))
+                      [[] #{}])
+                    first
+                    reverse
+                    vec)]
+    {:parameters merged}))
+
+(defn expand-qualified-keywords
```

Comment:
the name here is a bit misleading: it's more like `generate-swagger-spec`, right?

Reply:
This is what it's called in the equivalent spec and schema code. But I'm happy to change it if you don't mind the inconsistency.

Reply:
Oh okay, I'll need to look at the spec-tools and schema-tools versions and see if they make any sense.